docs/content/guides/developer/app-examples/e2e-counter.mdx

```diff
@@ -179,124 +185,196 @@ The UI design consists of two parts:
+    import { Box, Container, Flex, Heading } from "@radix-ui/themes";
+    import { useState } from "react";
+    import { CreateCounter } from "./CreateCounter";
+
+    function App() {
+      const currentAccount = useCurrentAccount();
+      const [counterId, setCounter] = useState(() => {
+        const hash = window.location.hash.slice(1);
+        return isValidSuiObjectId(hash) ? hash : null;
+      });
+
+      return (
+        <>
+          <Flex
+            position="sticky"
+            px="4"
+            py="2"
+            justify="between"
+            style={{
+              borderBottom: "1px solid var(--gray-a2)",
+            }}
+          >
+            <Box>
+              <Heading>dApp Starter Template</Heading>
+            </Box>
+
+            <Box>
+              <ConnectButton />
+            </Box>
+          </Flex>
+          <Container>
+            <Container
+              mt="5"
+              pt="2"
+              px="4"
+              style={{ background: "var(--gray-a2)", minHeight: 500 }}
+            >
+              {currentAccount ? (
+                counterId ? (
+                  null
+                ) : (
+                  <CreateCounter
+                    onCreated={(id) => {
+                      window.location.hash = id;
+                      setCounter(id);
+                    }}
+                  />
+                )
+              ) : (
+                <Heading>Please connect your wallet</Heading>
+              )}
+            </Container>
+          </Container>
+        </>
+      );
+    }
+
+    export default App;
+  ```
+</details>
 
-Next, call the `useSignAndExecuteTransaction` hook in your component, which provides a `mutate` function you can use in your `create` function:
-
-```tsx
-export function CreateCounter( { onCreated }: { onCreated: (id: string) => void; } ) {
-  const counterPackageId = useNetworkVariable("counterPackageId");
-  const suiClient = useSuiClient();
-  const { mutate: signAndExecute } = useSignAndExecuteTransaction({
-    execute: async ({ bytes, signature }) =>
-      await suiClient.executeTransactionBlock({
-        transactionBlock: bytes,
-        signature,
-        options: {
-          // Raw effects are required so the effects can be reported back to the wallet
-          showRawEffects: true,
-          showEffects: true,
-        },
-      }),
-  });	
-
-  function create() {
-    // TODO
-  }
-
-  return (
-    <Container>
-      <Button
-        size="3"
-        onClick={() => {
-          create();
-        }}
-      >
-        Create Counter
-      </Button>
-    </Container>
-  );
-
-}
-```
+This sets up your app to read the hash from the URL, and get the counter's ID if the hash is a valid object ID. Then, it either renders a `Counter` (which you define in the next step) if you have a counter ID, or the `CreateCounter` button from the previous step. When a counter is created, you update the URL, and set the counter ID.
```

Comment:
```suggestion
This sets up your app to read the hash from the URL, and get the counter's ID if the hash is a valid object ID. Then, if you have a counter ID, it renders a `Counter` (which you define in the next step). If you don't have a counter ID, then it renders the `CreateCounter` button from the previous step. When a counter is created, you update the URL, and set the counter ID.
```
Probably a better way to state this but I think breaking up the two options into separate sentences is more readable.